### PR TITLE
Fix editor (BE/ED) import from WoS files #1625

### LINKED ISSUE #1625: Editors do not show up on WoS import
# Bug description
Editors do not show up after WoS import:

## txt file content
Screenshot 2024-06-21 at 10 07 38

## imported file
Screenshot 2024-06-21 at 10 08 55

# Steps to reproduce
1. Go to https://www.webofscience.com/wos/woscc/full-record/WOS:000500696400004
2. Export as `plain text file`, select `full record`
3. Open file and see editors show up under `BE`
4. Import publication as WoS
5. See editors not show up

### PR DESCRIPTION
**Pull request by verheyenkoen**
FIXES #1625

### CODE REVIEW

File: cypress/fixtures/wos-000500696400004.txt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+C1 [Dubois, Hanne; Wullaert, Andy] Univ Ghent, Dept Internal Med, Ghent, Belgium.
+   [Dubois, Hanne; van Loo, Geert; Wullaert, Andy] VIB, VIB UGent Ctr Inflammat Res, Ghent, Belgium.
+   [Dubois, Hanne; Wullaert, Andy] Univ Ghent, Ghent Gut Inflammat Grp GGIG, Ghent, Belgium.
+   [van Loo, Geert] Univ Ghent, Dept Biomed Mol Biol, Ghent, Belgium.
+C3 Ghent University; Ghent University; Flanders Institute for Biotechnology
+   (VIB); Ghent University; Ghent University
+RP Wullaert, A (corresponding author), Univ Ghent, Dept Internal Med, Ghent, Belgium.; Wullaert, A (corresponding author), VIB, VIB UGent Ctr Inflammat Res, Ghent, Belgium.; Wullaert, A (corresponding author), Univ Ghent, Ghent Gut Inflammat Grp GGIG, Ghent, Belgium.
+EM andy.wullaert@ugent.be
+ED Doe, John
+   Dow, Jane
```
Comment (verheyenkoen) [inline cypress/fixtures/wos-000500696400004.txt:31]: Added these manually as I couldn't find a publication that has both.

File: backends/ris/decoder.go [hunk 2]
```diff
@@ -205,14 +205,11 @@ func mapRecord(r Record, p *models.Publication) {
...
-					firstName = nameParts[1]
-				}
-				c := models.ContributorFromFirstLastName(firstName, lastName)
-				p.Author = append(p.Author, c)
+				p.Author = append(p.Author, extractContributor(val))
+			}
+		case "BE", "ED":
+			for _, val := range v {
+				p.Editor = append(p.Editor, extractContributor(val))
 			}
```
Comment (verheyenkoen) [inline backends/ris/decoder.go:213]: Currently if both are present, both are imported, as opposed to authors where code `AF` is preferred over `AU`. Can be improved if necessary obviously.

Comment (mietcls) [reply backends/ris/decoder.go:213]: Thanks! I added some more information here, please let me know if this clarifies the question: https://github.com/ugent-library/biblio-backoffice/issues/1625#issuecomment-2256129725

File: backends/ris/decoder.go [hunk 3]
```diff
@@ -196,23 +196,27 @@ func mapRecord(r Record, p *models.Publication) {
...
+				p.Author = append(p.Author, extractContributor(val))
 			}
-			if k == "AF" {
-				p.Author = nil
+		case "AU":
+			// give preference to AF over AU
+			if _, ok := r["AF"]; !ok {
+				for _, val := range v {
+					p.Author = append(p.Author, extractContributor(val))
+				}
```
Comment (verheyenkoen) [inline backends/ris/decoder.go:208]: Make sure to carefully review and test this. I've refactored this and tested somewhat but not entirely sure if the new code is completely equivalent to the old code.

Comment (nics) [reply backends/ris/decoder.go:208]: looks good to me

**Review by nics — APPROVED** [review]

**Pull request merged**